app/db_functions/personal.py

```diff
@@ -74,11 +74,66 @@ async def is_exist_item_db(text: str, context: Context.id) -> bool:
     return await Item.exists().where((Item.text == text) & (Item.context == context))
 
 
+async def is_words_in_card_db(telegram_user_id: int, item_relation_id: ItemRelation.id) -> bool:
+    '''
+    The function checks if the user already has a pair of such items to study that are in item_relation.
+    '''
+    item_relation: ItemRelation = await get_item_relation_db(item_relation_id)
+    card: Card = await Card.objects().where(
+        (Card.user.telegram_user_id == telegram_user_id) &
+        (Card.item_relation.item_1.is_in((item_relation.item_1, item_relation.item_2))) &
+        (Card.item_relation.item_2.is_in((item_relation.item_1, item_relation.item_2)))
+    )
+    return bool(card)
+
+
+async def get_or_create_item_db(text: str, context: Context.id, author: User.id) -> Item:
+    item = await Item.objects().get_or_create((Item.text == text) & (Item.context == context),
+                                              defaults={'author': author,
+                                                        'context': context,
+                                                        'text': text})
+    return item
+
+
+async def get_or_create_user_db(data_telegram: aiogram.types.User) -> User:
+    user = await User.objects().get_or_create(User.telegram_user_id == data_telegram.id, defaults={
+        'telegram_language': data_telegram.language_code or "",
+        'user_name': data_telegram.username or "",
+        'first_name': data_telegram.first_name,
+        'last_name': data_telegram.last_name or ""})
+    return user
+
+
 async def get_context_id_db(name_alfa2: str) -> Context.id:
     context: Context = await Context.objects().get(Context.name_alfa2 == name_alfa2)
     return context.id
 
 
+async def get_item_relation_db(item_relation_id: ItemRelation.id) -> ItemRelation:
```

Comment:
```suggestion
async def get_item_relation_by_id_db(item_relation_id: ItemRelation.id) -> ItemRelation:
```
мені здається треба трохи змінити назву, бо у нас вже є get_item_relation_by_text
і там теж треба додати db наприкінці.

Reply:
Перейменувала.